golang/testsuite/testsuite_impl/wait_for_endpoint_availability_test/wait_for_endpoint_availability_test_.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020 - present Kurtosis Technologies LLC.
+ * All Rights Reserved.
+ */
+
+package wait_for_endpoint_availability_test
+
+import (
+	"github.com/kurtosis-tech/kurtosis-client/golang/networks"
+	"github.com/kurtosis-tech/kurtosis-client/golang/services"
+	"github.com/kurtosis-tech/kurtosis-libs/golang/lib/testsuite"
+	"github.com/kurtosis-tech/kurtosis-libs/golang/testsuite/services_impl/datastore"
+	"github.com/palantir/stacktrace"
+	"github.com/sirupsen/logrus"
+)
+const (
+	datastoreServiceId services.ServiceID = "datastore"
+
+	healthCheckUrlSlug = "health"
+	healthyValue       = "healthy"
+
+	waitForStartupTimeBetweenPolls = 1
+	waitForStartupMaxPolls = 15
+	waitInitialDelaySeconds = 1
+)
+
+
+type WaitForEndpointAbailabilityTest struct {
+	datastoreImage string
+}
+
+func NewWaitForEnpointAvailabilityTest(datastoreImage string) *WaitForEndpointAbailabilityTest {
```

Comment:
Another typo here: s/Enpoint/Endpoint/

Goland has a cool "Generate constructor" feature that I'd recommend using to avoid these - if you right-click a struct name, you can go Generate > Constructor

Reply:
Cool, I'll use that Goland command.